Remove full-text search escaping

geichelberger opened the pull request:
The search index uses `multi_match` for full-text search, which is based on match, and does not use the powerful query string syntax based on Apache Lucene query syntax. We must remove the escaping, or we will not get exact matches for keyword fields, like searching for the media package ID.

Fixup PR https://github.com/opencast/opencast/pull/6498

geichelberger committed:
Remove full-text search escaping

The search index uses `multi_match` for full-text search, which is based
on match, and does not use the powerful query string syntax based on
Apache Lucene query syntax. We must remove the escaping, or we will not
get exact matches for keyword fields, like searching for the media
package ID.

gregorydlogan reviewed (changes requested):
The whole `QueryPreprocessor` class is now unused, remove this too?

geichelberger committed:
Remove unused QueryPreprocessor class

The query preprocessor class is obsolete as we switch to a search method
that does not support a query syntax.
